### DISCUSSION TITLE
When i can get window size?

### DISCUSSION THREAD

**Discussion by xtuzy** [dotnet/maui#7586] [Q&A]
I need layout according to window size, such as at ipad, app can multiple task, that mean window not full screen, i need detect window size and use different layout with full screen. 
But i don't know when to get correct window size, when i try get MainPage size at constructor, the Width and Height are not correct. I found have SizeChang event, it will be loaded multiple when window show at Windows, so i need set layout at SizeChange event handler method?
I hope have a best practice for develop maui app🤔

**Comment by hallon-heyman** [comment]
This might be too late to respond, but if anyone else is googling this as I was: The normal way to do this would be through `DeviceDisplay.Current.MainDisplayInfo ` but at the time of this post, this is bugged and won't be fixed during 6.0, see https://github.com/dotnet/maui/issues/6976

**Reply by Moreno-Gentili** [reply]
> DeviceDisplay.Current.MainDisplayInfo 

That would return the display size. I'd like to get the **app window** size via a BindableProperty, so that when user resizes the window I can modify my layout accordingly.
I could create my own BindableProperty in the Shell but I can't since [Shell.SizeChanged is not firing](https://github.com/dotnet/maui/issues/7227).

**Reply by brianlellis** [reply]
This answer is further misleading in that MainDisplayInfo is only for the main display of the device. If you have multiple displays this value only returns the value of the main display despite where the application window is living on the OS.

**Comment by brianlellis** [comment]
In Mac Catalyst (Desktop project, but the API should behave the same on IOS) I have had luck with `Application.Current.MainPage.Width/Height`.

The Windows UI looks to handle rendering and layout measurement events differently than Mac. That said here is the basics of layout phases.

Key Phases in the XAML Layout Lifecycle:
  1.	Initialization Phase
  2.	Measuring Phase (Measure Pass)
  3.	Arranging Phase (Arrange Pass)
  4.	Rendering Phase
  
  
 There's several places to hook into during these phases which should be able to get you dimension values.
 
 `protected override void OnSizeAllocated(double width, double height)`
 
 and
 
 ```c#
 protected override async void OnAppearing()
{
    base.OnAppearing();
    await Task.Delay(100); // Short delay to allow layout measurement to complete, THIS IS NOT IDEAL!

    var width = this.Width;
    var height = this.Height;
    Console.WriteLine($"MainPage Width: {width}, Height: {height}");
}
```

It should be noted that my app is a fully drawn SkiaSharp application so I'm accessing `Application.Current.MainPage.Width/Height` during `OnPaintSurface()`.
